src/terminate.rs

```diff
@@ -37,13 +37,17 @@ impl Module {
                 insts.push(Instruction::GlobalSet(fuel_global));
             };
 
-            let mut new_insts = Vec::with_capacity(code.instructions.len() * 2);
+            let instrs = match &mut code.instructions {
+                Instructions::Generated(list) => list,
+                Instructions::Arbitrary(_) => continue,
+            };
+            let mut new_insts = Vec::with_capacity(instrs.len() * 2);
 
             // Check fuel at the start of functions to deal with infinite
             // recursion.
             check_fuel(&mut new_insts);
 
-            for inst in mem::replace(&mut code.instructions, vec![]) {
+            for inst in mem::replace(instrs, vec![]) {
```

Comment:
This can't ever be called on `MaybeInvalidModule`s, right? And therefore we shouldn't ever see `Instructions::Arbitrary` here, right? If I understand that correctly, then lets assert that here, rather than `continue`.